src/intrprtr.c

```diff
@@ -322,8 +314,7 @@ ExecStatus IntrEnd (
     }
 
     // switch back to the old state
-    STATE(StackObj)  = ELM_PLIST(STATE(IntrState), 2);
-    STATE(IntrState) = ELM_PLIST(STATE(IntrState), 1);
+    STATE(StackObj) = PopPlist(STATE(IntrState));
```

Comment:
Note that `PopPlist` not just removes the element from the list; it also overwrites the slot in the plist that used to store the pop'ed element by zero, which ensures that it can be garbage collected right away (instead of leaving a "fake" reference to it inside the plist).